components/sync/driver/brave_sync_profile_sync_service.cc

```diff
@@ -9,13 +9,22 @@
 #include <vector>
 
 #include "base/logging.h"
+#include "base/task/post_task.h"
 #include "brave/components/brave_sync/crypto/crypto.h"
 #include "brave/components/sync/driver/brave_sync_auth_manager.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/browser/sync/device_info_sync_service_factory.h"
 #include "components/prefs/pref_service.h"
+#include "components/sync_device_info/device_info_sync_service.h"
+#include "components/sync_device_info/device_info_tracker.h"
+#include "components/sync_device_info/local_device_info_provider.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
 
 namespace syncer {
 
-BraveProfileSyncService::BraveProfileSyncService(InitParams init_params)
+BraveProfileSyncService::BraveProfileSyncService(InitParams init_params,
+                                                 Profile* profile)
```

Comment:
This is layer violation.
Do not include `//chrome/browser/*` headers in `//brave/components/*`.
We can observe `OnDeviceInfoChange` in `ProfileSyncServiceHelper` class which we don't have right now.

Reply:
Fixed at https://github.com/brave/brave-core/pull/6732/commits/16066dea3243dce2d5d505969b0bd5252efb75b0 , with instructions at https://www.chromium.org/developers/design-documents/cookbook and example of https://github.com/chromium/chromium/commit/8d4cc15e95bdce1ec41b68c2b1cc5685642c39e0